packages/core/src/components/fields/SchemaField.js

```diff
@@ -362,7 +362,9 @@ function SchemaFieldRender(props) {
             onBlur={props.onBlur}
             onChange={props.onChange}
             onFocus={props.onFocus}
-            options={schema.anyOf}
+            options={schema.anyOf.map(_schema =>
+              retrieveSchema(_schema, rootSchema, formData)
```

Comment:
I guess this is technically a breaking change, since schemas passed to `AnyOfField` are dereferenced. This should be okay, though, given that otherwise, the non-dereferenced schemas would be useless because the `rootSchema` isn't passed to fields anyway.

Reply:
Yeah I agree and since `retrieveSchema` is idempotent then it should be okay so I figured it was best to do this as high up as possible (also based on a comment from you earlier).

Let me know if you think this can stay

Reply:
Yeah, this can stay